signer/src/block_observer.rs

```diff
@@ -489,18 +490,21 @@ impl<C: Context, B> BlockObserver<C, B> {
     /// Update the sBTC peg limits from Emily
     async fn update_sbtc_limits(&self) -> Result<(), Error> {
         let limits = self.context.get_emily_client().get_limits().await?;
-        let max_mintable = match limits.total_cap() {
-            Amount::MAX_MONEY => Amount::MAX_MONEY,
-            total_cap => {
-                let sbtc_supply = self
-                    .context
-                    .get_stacks_client()
-                    .get_sbtc_total_supply(&self.context.config().signer.deployer)
-                    .await?;
-                // The maximum amount of sBTC that can be minted is the total cap
-                // minus the current supply.
-                total_cap.checked_sub(sbtc_supply).unwrap_or(Amount::ZERO)
-            }
+
+        let max_mintable = if limits.total_cap_exists() && self.is_token_deployed().await? {
```

Comment:
Just a thought, but what if we move the `sbtc_contracts_deployed` field from the `TxCoordinatorEventlLoop` to the `SignerState` so that we can query it from here to avoid an extra call to the node?

https://github.com/stacks-network/sbtc/blob/main/signer/src/transaction_coordinator.rs#L1480

Reply:
We'd have to also check that the other smart contracts being deployed as well, since this only checks for the token contract. The remaining three smart contracts get submitted after the token one, and they can be confirmed much later.

Reply:
Oh wait, but this could work if the `tx-coordinator` sets some state variable for each contract when it checks (which is what you had in mind to begin with).

Reply:
Moved to the state -- I've used a single flag, set by the coordinator once all the smart contracts are confirmed as deployed.